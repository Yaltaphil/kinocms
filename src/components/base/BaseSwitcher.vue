/** Bootstrap switch component for Vue2 - yaltaphil */
<template>
    <div class="custom-control custom-switch custom-switch-on-success">
        <input
            id="customSwitch"
            type="checkbox"
            class="custom-control-input"
            :checked="checked"
            @change="$emit('change', $event.target.checked)"
        />
        <label class="custom-control-label" for="customSwitch"> <slot /></label>
    </div>
</template>

<script>
export default {
    name: "BaseSwitcher",

    // vue2 v-model binding
    model: {
        prop: "value",
        event: "change",
    },
    props: {
        checked: Boolean,
    },
};
</script>
