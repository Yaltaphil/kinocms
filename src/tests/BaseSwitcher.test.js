import { mount } from "@vue/test-utils";
import BaseSwitcher from "@/components/base/BaseSwitcher.vue";

const someText = "lorem10";

describe("Testing BaseSwitcher Component", () => {
    const wrapper = mount(BaseSwitcher, {
        propsData: { checked: false },
        slots: {
            default: someText,
        },
    });
    test("This switsher exists and visible:", () => {
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.isVisible()).toBeTruthy();
    });

    test("Props comes to switcher:", () => {
        expect(wrapper.vm.checked).toBeDefined();
        expect(wrapper.vm.checked).toBe(false);
    });

    test("Slot works ok:", () => {
        expect(wrapper.find("label").text()).toMatch(someText);
    });

    test("Switcher switches:", () => {
        expect(wrapper.find("input").setChecked()).toBeTruthy();
    });

    test("Switcher emits the event:", () => {
        wrapper.find("input").trigger("change");
        expect(wrapper.emitted("change")).toBeTruthy();
    });
});
