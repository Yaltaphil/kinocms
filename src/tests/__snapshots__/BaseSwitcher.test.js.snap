// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BaseSwitcher Component Switcher matches snapshot: 1`] = `
<div
  class="form-group"
>
  <div
    class="
            custom-control
            custom-switch
            custom-switch-off-danger
            custom-switch-on-success
        "
  >
    <input
      class="custom-control-input"
      id="1"
      type="checkbox"
    />
     
    <label
      class="custom-control-label"
      for="1"
    >
      lorem10
    </label>
  </div>
</div>
`;
